### ISSUE TITLE
bug: kill_by_port in stop() can SIGKILL the calling process

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/redis-server-wrapper#76]
## Problem

`RedisServerHandle::stop()` runs `kill_by_port` as a final safety net (step 4). This calls `lsof -ti :{port}` and kills every process it finds. If the calling process has an open TCP connection to the same port (e.g. a Redis client connection), `lsof` returns the calling process's PID and `stop()` kills it.

This reliably reproduces in integration tests where:
1. A test starts a Redis server on port N via `RedisServer::new().port(N).start().await`
2. The test opens a client connection to port N
3. The test calls `handle.stop()` or the handle drops
4. `kill_by_port(N)` finds the test binary (which has a socket to port N) and kills it

The result is `signal: 9, SIGKILL: kill` on the test binary after all tests pass.

## Reproduction

```rust
#[tokio::test]
async fn repro() {
    let server = RedisServer::new().port(6401).start().await.unwrap();
    
    // Open a client connection to the server
    let _conn = tokio::net::TcpStream::connect("127.0.0.1:6401").await.unwrap();
    
    // This kills the test binary via kill_by_port
    drop(server);
}
```

## Suggested fix

`kill_by_port` should filter out the calling process's own PID before killing. Something like:

```rust
pub fn kill_by_port(port: u16) {
    let my_pid = std::process::id().to_string();
    // ... lsof ...
    for pid in pids {
        if pid != my_pid {
            // kill
        }
    }
}
```

Alternatively, `kill_by_port` could be limited to processes that are *listening* on the port (not just connected to it) by using `lsof -ti :{port} -sTCP:LISTEN`.

## Workaround

For tests that need to stop/restart a server while holding client connections, use direct `redis-cli SHUTDOWN NOSAVE` instead of the handle's `stop()`, or call `handle.detach()` before the handle drops (but this leaks the server process).
